client/scala/src/main/scala/ai/verta/blobs/dataset/AtlasHiveDatasetBlob.scala

```diff
@@ -0,0 +1,150 @@
+        }
+      }
+      case _ => Failure(throw new IllegalArgumentException(s"unknown type ${value.getClass.toString}"))
+    }
+
+  // get the first entity.
+  // equivalent to table_obj["entities"][0]
+  private def extractEntity(fieldsMap: Map[String, JValue]): Try[Map[String, JValue]] = {
+    fieldsMap.get("entities") match {
+      case Some(JArray(elements)) => elements.headOption match {
+        case Some(JObject(entityFields)) => Success(entityFields.map(f => (f.name, f.value)).toMap)
+        case Some(other) => Failure(new IllegalArgumentException(s"unknown type ${other.getClass.toString}")) // does this case happen?
+        case None => Failure(new IllegalArgumentException(s"no entity found."))
+      }
+      case Some(other) => Failure(new IllegalArgumentException(s"unknown type ${other.getClass.toString}"))
+      case None => Failure(new IllegalArgumentException("\"entities\" field not found."))
+    }
+  }
+
+  private def checkEntityType(entityMap: Map[String, JValue]): Try[Unit] =
+    if (getEntityType(entityMap) == "hive_table")
+      Success(())
+    else
+      Failure(new IllegalArgumentException("Atlas dataset currently supported only for Hive tables."))
+
+  private def getEntityType(entityMap: Map[String, JValue]) =
+    entityMap.get("typeName").map(JsonConverter.fromJsonString).get
+
+  // extract a certain field from a json map, which in turn is another map
+  private def getSubMap(map: Map[String, JValue], key: String): Try[Map[String, JValue]] =
+    map.get(key) match {
+      case Some(JObject(fields)) => Success(fields.map(f => (f.name, f.value)).toMap)
+      case Some(other) => Failure(new IllegalArgumentException(s"unknown type ${other.getClass.toString}"))
+      case None => Failure(new IllegalArgumentException(f"key ${key} is not in map"))
+    }
+
+    private def getAttributes(
+      entityMap: Map[String, JValue],
+      attributesMap: Map[String, JValue],
+      dbRelationshipAttributesMap: Map[String, JValue]
+    ): Map[String, ValueType] = {
+      /** TODO: Support more attributes, namely the one in list forms: load_queries, col_names */
+
+      val tableName: String = attributesMap.get("name").map(JsonConverter.fromJsonString).get
+      val databaseName: String = dbRelationshipAttributesMap.get("displayText").map(JsonConverter.fromJsonString).get
+      val createdTime: BigInt = entityMap.get("createTime").map(JsonConverter.fromJsonInteger).get
+      val updatedTime: BigInt = entityMap.get("updateTime").map(JsonConverter.fromJsonInteger).get
+
+      Map[String, ValueType](
+        "type" -> "hive_table", // only supports hive table.
+        "table_name" -> tableName,
+        "database_name" -> databaseName,
+        "created_time" -> createdTime,
+        "updated_time" -> updatedTime
+      )
+    }
+
+    private def getTags(entityMap: Map[String, JValue]): List[String] = {
```

Comment:
I'm assuming this was copied from the python client.

Reply:
Yep.